elasticsearch_plugin - Show STDERR on module failures.

kivilahtio opened the pull request:
##### SUMMARY
I tried to install a ES plugin without `become: yes`and found after debugging the module that the module failed due to permission issues.

The only error message I got was
  Is analysis-icu a valid plugin name?

That was strange considering I followed the example documentation by the letter.

I found out that when this module fails, it hides the real reason for failure.

This patch replaces the generic error with more meaningful diagnostics.


##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
elasticsearch_plugin

##### ANSIBLE VERSION
```
ansible 2.5.3
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

Commit:
elasticsearch_plugin - Show STDERR on module failures.

I tried to install a ES plugin without
  become: yes
and found after debugging the module that the module failed ude to permission issues.

The only error message I got was
  Is analysis-icu a valid plugin name?

That was strange considering I followed the example documentation by the letter.

I found out that when this module fails, it hides the real reason for failure.

This patch replaces the generic error with more meaningful diagnostics.

samdoran commented:
This looks good. Please create a changelog fragment to go along with this change. See [fragments](https://github.com/ansible/ansible/tree/stable-2.6/changelogs/fragments) for examples.

kivilahtio committed:
elasticsearch_plugin - Show STDERR on module failures. Changelog fragment

 samdoran commented 2 days ago

This looks good. Please create a changelog fragment to go along with this change. See fragments for examples.

kivilahtio commented:
Ok. Added a followup.
